tests/virt_autotest/reset_partition.pm

```diff
@@ -0,0 +1,112 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package reset_partition;
+# Summary: reset_partition: Ensure and reset hard disk partition for virt_autotest.
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use virt_utils;
+use ipmi_backend_utils;
+use Utils::Architectures;
+use utils qw(zypper_call);
+use version_utils 'is_sle';
+use virt_autotest::utils qw(is_xen_host);
+
+sub reset_partition {
+    my ($libvirt_disk) = @_;
+    my $new_disk       = "";
+    my $virt_point     = "/var/lib/libvirt/";
+
+    # Install requested pkgs
+    my @pkgs = qw/xfsprogs coreutils util-linux/;
+    zypper_call "in @pkgs";
+
+    # Get some debug info about hard disk topology
+    record_info("Report file system disk space", script_output("df"));
+    record_info("Find mounted filesystem",       script_output("findmnt"));
+
+    # Check with all existed Hard disks
+    my @disks = split(/\n/, script_output("lsblk -n -l -o NAME -d -e 7,11"));
+    my $dev   = "/dev/";
+    foreach my $disk (@disks) {
+        assert_script_run('file ' . $dev . $disk);
+    }
+
+    # Assign a New FULL Disk for /var/lib/libvirt
+    foreach (0 .. $#disks) {
```

Comment:
For this logic, no foreach is needed, you actually just need:
        # Use the first disk or the second(if libvirt image takes the first disk already) as the new disk
        if ($libvirt_disk eq $disks[0]) {
            $new_disk = $dev . $disks[1];
        } else {
            $new_disk = $dev . $disks[0]; 
        }

Reply:
sure, had updated code already.